pkg/controller/argocd/service.go

```diff
@@ -307,13 +307,39 @@ func (r *ReconcileArgoCD) reconcileRedisService(cr *argoprojv1a1.ArgoCD) error {
 	return r.client.Create(context.TODO(), svc)
 }
 
+func ensureAutoTLSAnnotation(cr *argoprojv1a1.ArgoCD, svc *corev1.Service) bool {
+	autoTLSAnnotationName := ""
+	if cr.Spec.Repo.AutoTLS == "openshift" {
+		autoTLSAnnotationName = "service.beta.openshift.io/serving-cert-secret-name"
+	}
+	if autoTLSAnnotationName != "" {
+		if svc.Annotations == nil {
+			svc.Annotations = make(map[string]string)
+		}
+		val, ok := svc.Annotations[autoTLSAnnotationName]
+		if !ok || val != "argocd-repo-server-tls" {
```

Comment:
Hmm ...  we have a const ArgoCDRepoServerTLSSecretName.   Why not using it?

Reply:
Good catch, thank you. Will update.

Reply:
Updated.